arithmetization/src/main/java/net/consensys/linea/blockcapture/snapshots/StorageSnapshot.java

```diff
@@ -19,17 +19,36 @@
 
 import org.apache.tuweni.units.bigints.UInt256;
 import org.hyperledger.besu.datatypes.Address;
+import org.hyperledger.besu.evm.account.Account;
 import org.hyperledger.besu.evm.worldstate.WorldView;
 
 public record StorageSnapshot(String address, String key, String value) {
   public static Optional<StorageSnapshot> from(
       Address address, UInt256 key, final WorldView world) {
-    return Optional.ofNullable(world.get(address))
-        .map(
-            account ->
-                new StorageSnapshot(
-                    address.toHexString(),
-                    key.toHexString(),
-                    account.getStorageValue(key).toHexString()));
+    // Lookup account that was touched
+    Account account = world.get(address);
+    // Check account *really* exists from a storage perspective.  This is important to distinguish
+    // for accounts which are created *during* the conflation.  Such accounts may have technically
+    // existed before the conflation (e.g. they had a non-zero balance) but could still have been
+    // "created" during the conflation.  In such case, this snapshot would be simply assigning 0x0
+    // to the given storage locations.  However, we don't want to create a storage snapshot in such
```

Comment:
Does taking a storage snapshot insert a storage value ?

Reply:
Also, at least according to the current EYP, storage does not matter when it comes to checking the possibility of doing a CREATE2

<img width="757" alt="image" src="https://github.com/user-attachments/assets/5f6c656c-10f6-461e-82bf-a7747480cda3">

Reply:
> Also, at least according to the current EYP

Yeah, but at the end of the day, we are not measuring against the EYP.  We are comparing against execution on MAINNET using BESU.  What I'm looking at is what BESU does.  If that deviates from the EYP ... well, we still have to follow BESU.

However, I agree that it does seem wierd and I can follow up with BESU to see why they do it that way (or even it could be a bug).

Reply:
> Does taking a storage snapshot insert a storage value ?

Yes, in the replay environment it does.  See `ReplayExecutionEnvironment.initWorld()`.  One option might be to drop `StorageSnapshots` which assign `0x0` at the beginning of the conflation.  It has the same effect.  Maybe you prefer this?

Reply:
But does taking a storage snapshot change the world state ? If so that is a massive problem.

Reply:
In any case we do want to define storage snapshots even if the target account has empty storage or doesn't even exist yet. Simply because it's possible in Ethereum and we will want to test that, see https://github.com/Consensys/linea-tracer/issues/1205